lib/ClangImporter/ClangImporter.cpp

```diff
@@ -448,9 +448,16 @@ getNormalInvocationArguments(std::vector<std::string> &invocationArgStrs,
   invocationArgStrs.insert(invocationArgStrs.end(), {
       // Enable modules
       "-fmodules",
-      "-Werror=non-modular-include-in-framework-module",
-      "-Xclang", "-fmodule-feature", "-Xclang", "swift",
+      "-Xclang", "-fmodule-feature", "-Xclang", "swift"
+  });
+  // Don't enforce strict rules when inside the debugger to work around search
+  // path problems caused by a module existing in both the build/install
+  // directory and the source directory.
+  if (!importerOpts.DebuggerSupport)
+    invocationArgStrs.push_back(
+        "-Werror=non-modular-include-in-framework-module");
```

Comment:
Don't worry about keeping these in the same order. It's okay to put this after the fixed set of normal arguments.

Reply:
I thought there was value in grouping all module-related options together.

Reply:
Hm, okay, but then I'd rather move the whole section below the completely unconditional and uncategorized settings.